docs/architecture-and-design.md

```diff
@@ -386,6 +389,20 @@ Most [Intents](#intents) are easy to break down into  [Primitives](#primitives),
 
 In order for a robot to move to the desired destination of a `MoveIntent`, the Navigator will use various path-planning algorithms to find a path across the field that does not collide with any robots or violate any restrictions set on the `MoveIntent`. The Navigator then translates this path into a series of `MovePrimitives`, which are sent to the robot sequentially so that it follows the planned path across the field.
 
+### Path Manager
+The `Path Manager` is responsible for generating a set of paths that don't collide. It is given a set of path objectives and path planner and it will generate paths using the given path planner and arbitrate between paths to prevent collisions. One approach is to add extra obstacles to cause the path planner to generate paths that avoid a potential collisions; this is what `VelocityObstaclePathManager` implements. One disadvantage of this approach is that it does not account for when robots will occupy particular positions on the path. (Trajectory planning or use of the Minkowski space are possible ideas to account for time.)
+
+### Path Planner
+The `Path Planner` is responsible for path planning a single robot around a single set of obstacles from a given start to a given destination. Note that the interface of the path planner is intentionally decoupled from the rest of AI, so that it doesn't rely on the representation of [Intents](#intents) or [World](#world).
+
+#### Theta Star Path Planner
+The `Theta Star Path Planner` is a path planner that determines a series of points from the start and destination, such that none of the points are in an obstacle and none of the line segments between adjacent points cross an obstacle, optimizing for the shortest total path length. It uses a grid of points to represent the field and uses graph exploration to connect the start and destination. As it explores each point, it checks against each of the given obstacles to see if that point is inside. 
+A Star is a similar algorithm that does the same graph exloration as Theta Star. The key difference is that Theta Star will try to connect each newly explored point to the previous point on the path via line of sight by checking that the line of sight doesn't cross an obstacle. This means the path isn't confined to moving along right angles of the grid as in A Star and is free to move at any angle. Why not smooth the path afterwards? Looking for line of sight allows the algorithm to accurately evaluate the smoothed path length as it goes. This avoids cases with A Star where the right-angled path it generates might be the shortest when compare against other right-angled paths, but once smoothed, it isn't the shortest when compared to other smoothed paths.
```

Comment:
Suggest `A*` instead of `A Star`, just because that is how it is typically denoted in literature. The reason I'm not also suggesting that for `Theta*` is that spelling that out fully helps link people back to our codebase, whereas with A* we only really need them make connections to external resources.

Same goes for other instances of `A Star` here.